cmake/macos/xcode.cmake

```diff
@@ -85,12 +85,10 @@ set(CMAKE_XCODE_ATTRIBUTE_LINKER_DISPLAYS_MANGLED_NAMES[variant=Debug] YES)
 set(CMAKE_XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_ARC NO)
 # Disable strict aliasing
 set(CMAKE_XCODE_ATTRIBUTE_GCC_STRICT_ALIASING NO)
-# cmake-format: off
-# Re-enable once the bad goto statements are fixed
-# set(CMAKE_XCODE_ATTRIBUTE_GCC_C_LANGUAGE_STANDARD c99)
-# cmake-format: on
-# Set C++ language default to c++17
-set(CMAKE_XCODE_ATTRIBUTE_CLANG_CXX_LANGUAGE_STANDARD c++17)
+# Set C language default to GNU flavour C17
+set(CMAKE_XCODE_ATTRIBUTE_GCC_C_LANGUAGE_STANDARD gnu17)
+# Set C++ language default to GNU flavour C++17
+set(CMAKE_XCODE_ATTRIBUTE_CLANG_CXX_LANGUAGE_STANDARD gnu++17)
```

Comment:
For the record, I would very much like if we stuck to ISO C17 and C++17. However, since apparently Xcode does not list C17 as an option, we're using GNU flavors here (which apparently we were defaulting to on macOS anyway). If that's the case, we should consider using `-pedantic` and `-Wpedantic` so that devs do not accidentally make use of a GNU extension in cross-platform code that will not work elsewhere. That can be done separately, so I won't block on this here.

Reply:
Xcode does support c17 and c++17, however c17 is stricter than gnu17 and would require us to fix code in some places. Also CMake defaults to gnu17 when the language standard is set via `CMAKE_C_STANDARD`.

Reply:
If this is something that must be fixed in parts, that's fine with me. I just think that ISO C/C++ should be the longterm goal here for compatibility across our supported compilers and avoiding accidental extension usage.